lib/dns_cluster.ex

```diff
@@ -90,7 +91,8 @@ defmodule DNSCluster do
         {:ok, state, {:continue, :discover_ips}}
 
       {:ok, other} ->
-        raise ArgumentError, "expected :query to be a string, got: #{inspect(other)}"
+        raise ArgumentError,
+              "expected :query to be a string or list of strings, got: #{inspect(other)}"
```

Comment:
note that we don't validate that the list contains only strings. Happy to add that if we think its necessary (seems like its not to me)